embedding/cpp/flutter_engine.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "include/flutter_engine.h"
+
+#include "tizen_log.h"
+#include "utils.h"
+
+std::unique_ptr<FlutterEngine> FlutterEngine::Create(
+    const std::optional<std::string>& dart_entrypoint,
+    const std::optional<std::vector<std::string>>& dart_entrypoint_args) {
+  return std::move(FlutterEngine::Create(
+      "../res/flutter_assets", "../res/icudtl.dat", "../lib/libapp.so",
+      dart_entrypoint, dart_entrypoint_args));
+}
+
+std::unique_ptr<FlutterEngine> FlutterEngine::Create(
+    const std::string& assets_path, const std::string& icu_data_path,
+    const std::string& aot_library_path,
+    const std::optional<std::string>& dart_entrypoint,
+    const std::optional<std::vector<std::string>>& dart_entrypoint_args) {
+  FlutterEngine* engine =
+      new FlutterEngine(assets_path, icu_data_path, aot_library_path,
+                        dart_entrypoint, dart_entrypoint_args);
+  if (engine->engine_) {
+    return std::unique_ptr<FlutterEngine>(engine);
+  } else {
+    delete engine;
+    return nullptr;
+  }
+}
+
+FlutterEngine::FlutterEngine(
+    const std::string& assets_path, const std::string& icu_data_path,
+    const std::string& aot_library_path,
+    const std::optional<std::string>& dart_entrypoint,
+    const std::optional<std::vector<std::string>>& dart_entrypoint_args) {
+  FlutterDesktopEngineProperties engine_prop = {};
+  engine_prop.assets_path = assets_path.c_str();
+  engine_prop.icu_data_path = icu_data_path.c_str();
+  engine_prop.aot_library_path = aot_library_path.c_str();
+
+  // Read engine arguments passed from the tool.
+  Utils::ParseEngineArgs(&engine_arguments_);
```

Comment:
This function is now only used here so it might be moved from `utils.h` to this class.

Reply:
I moved `ParseEngineArgs` to this class.

Reply:
```cpp
std::vector<std::string> engine_args = ParseEngineArgs();

std::vector<const char*> switches;
for (...) {
```
or even
```cpp
for (const std::string& arg : ParseEngineArgs()) {
  ...
```
seems more appropriate in the current structure.

Reply:
I chose the first proposal.